numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -1173,10 +1174,29 @@ NpyIter_NestedIters(PyObject *NPY_UNUSED(self),
     return NULL;
 }
 
+
 static void
 npyiter_dealloc(NewNpyArrayIterObject *self)
 {
     if (self->iter) {
+        if (npyiter_has_writeback(self->iter)) {
+            if (PyErr_WarnEx(PyExc_RuntimeWarning,
+                    "Temporary data has not been written back to one of the "
+                    "operands. Typically nditer is used as a context manager "
+                    "otherwise 'close' must be called before reading iteration "
+                    "results.", 1) < 0) {
```

Comment:
Do we have a solution to #10956 yet? Is it guaranteed that this warning will never appear if `out` is allocated by the iterator?

Reply:
@mattip This seems to be the only outstanding issue here that needs to be resolved.

Reply:
I am not sure #10956 is an issue. See my last comment there. It is more a discussion about what is meant by operands. In any case, if `out` is allocated by the iterator it will never have writeback semantics and so the warning will not appear.

Reply:
> In any case, if out is allocated by the iterator it will never have writeback semantics and so the warning will not appear.

Great, this is what I was worried about